Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -881,10 +881,13 @@ void GraphicsContextSkia::drawPattern(NativeImage& nativeImage, const FloatRect&
     if (!makeGLContextCurrentIfNeeded())
         return;
 
-    FloatRect rect(tileRect);
-    rect.moveBy(phase);
     SkMatrix phaseMatrix;
-    phaseMatrix.setTranslate(rect.x(), rect.y());
+    // Adjust phase based on tile offset and spacing.
+    double phaseOffsetX = phase.x() + tileRect.x() * patternTransform.a();
+    double phaseOffsetY = phase.y() + tileRect.y() * patternTransform.d();
+    phaseOffsetX -= std::trunc(phaseOffsetX / (tileRect.width() * patternTransform.a() + spacing.width())) * (tileRect.width() * patternTransform.a() + spacing.width());
+    phaseOffsetY -= std::trunc(phaseOffsetY / (tileRect.height() * patternTransform.d() + spacing.height())) * (tileRect.height() * patternTransform.d() + spacing.height());
```

Comment:
I don't think we need this, it was done in Cairo code due to the surface size limitation, but that's cairo specific.